src/rebar_raw_resource.erl

```diff
@@ -254,42 +272,72 @@ download(Dest, {?RTYPE, Spec, Opts}, State) ->
             Err
     end.
 
--spec lock(Path :: rsrc_dir(), Spec :: this_spec())
-        -> rebar_lock() | no_return().
+-spec download(Dest :: rsrc_dir(),
+               AppInfo :: rebar_app_info:t(),
+               ResourceState :: term(),
+               State :: rebar_state())
+              -> {'ok', term()} | rebar_err().
+%
+% Download the specified resource using its underlying handler.
+%
+download(TmpDir, AppInfo, _ResourceState, RebarState) ->
+    download(TmpDir, rebar_app_info:source(AppInfo), RebarState).
+
+-spec lock(Path :: rsrc_dir(),
+           Spec :: this_spec())
+          -> rebar_lock() | no_return();
+          (AppInfo :: rebar_app_info:t(),
+           ResourceState :: term())
+          -> rebar_lock() | no_return().
 %
 % Pass through to the underlying resource handler.
 % Note that the callback doesn't allow an error tuple to be returned, so an
 % exception is our only option if we can't look up the mapping.
 %
-lock(Path, {?RTYPE, Loc, #mod_ref{res = Res, ref = Prev} = Rec}) ->
+%% pre-rebar3 3.7.0 format
+lock(Path, Source) when not(erlang:is_tuple(Path)) ->
```

Comment:
Yeah, I used this guard because the one in their example is useless, because it guards on source being a tuple, but AFAICT it's a tuple in the old style and the new style, so... yeah. This one actually verifies that the first arg isn't a record (which is the new first arg in the current resource format).